applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -243,79 +242,23 @@ void UPwSmallStrainElement<TDim, TNumNodes>::FinalizeSolutionStep(const ProcessI
         mConstitutiveLawVector[integration_point]->FinalizeMaterialResponseCauchy(ConstitutiveParameters);
         mStateVariablesFinalized[integration_point] = mConstitutiveLawVector[integration_point]->GetValue(
             STATE_VARIABLES, mStateVariablesFinalized[integration_point]);
-
-        if (rCurrentProcessInfo[NODAL_SMOOTHING])
-            this->SaveGPStress(StressContainer, mStressVector[integration_point], integration_point);
     }
 
-    if (rCurrentProcessInfo[NODAL_SMOOTHING]) this->ExtrapolateGPValues(StressContainer);
+    if (rCurrentProcessInfo[NODAL_SMOOTHING]) this->ExtrapolateGPValues();
 
     KRATOS_CATCH("")
 }
 
 template <unsigned int TDim, unsigned int TNumNodes>
-void UPwSmallStrainElement<TDim, TNumNodes>::SaveGPStress(Matrix&       rStressContainer,
-                                                          const Vector& StressVector,
-                                                          unsigned int  GPoint)
+void UPwSmallStrainElement<TDim, TNumNodes>::ExtrapolateGPValues()
 {
     KRATOS_TRY
 
-    for (unsigned int i = 0; i < StressVector.size(); ++i) {
-        rStressContainer(GPoint, i) = StressVector[i];
-    }
-
-    /* INFO: (Quadrilateral_2D_4 with GI_GAUSS_2)
-     *
-     *                      |S0-0 S1-0 S2-0|
-     * rStressContainer =   |S0-1 S1-1 S2-1|
-     *                      |S0-2 S1-2 S2-2|
-     *                      |S0-3 S1-3 S2-3|
-     *
-     * S1-0 = S[1] at GP 0
-     */
-
-    KRATOS_CATCH("")
-}
-
-template <unsigned int TDim, unsigned int TNumNodes>
-void UPwSmallStrainElement<TDim, TNumNodes>::ExtrapolateGPValues(const Matrix& StressContainer)
-{
-    KRATOS_TRY
-
-    GeometryType&               r_geometry = this->GetGeometry();
-    const double&               area       = r_geometry.Area(); // In 3D this is volume
-    array_1d<Vector, TNumNodes> nodal_stress_vector;            // List with stresses at each node
-    array_1d<Matrix, TNumNodes> nodal_stress_tensor;
-
-    auto const StressTensorSize = this->GetStressStatePolicy().GetStressTensorSize();
-    for (unsigned int iNode = 0; iNode < TNumNodes; ++iNode) {
-        nodal_stress_vector[iNode].resize(this->GetStressStatePolicy().GetVoigtSize());
-        nodal_stress_tensor[iNode].resize(StressTensorSize, StressTensorSize);
-    }
-
-    BoundedMatrix<double, TNumNodes, TNumNodes> extrapolation_matrix;
-    this->CalculateExtrapolationMatrix(extrapolation_matrix);
-
-    Matrix AuxNodalStress;
-    AuxNodalStress.resize(TNumNodes, this->GetStressStatePolicy().GetVoigtSize());
-    noalias(AuxNodalStress) = prod(extrapolation_matrix, StressContainer);
-
-    /* INFO:
-     *
-     *                  |S0-0 S1-0 S2-0|
-     * AuxNodalStress = |S0-1 S1-1 S2-1|
-     *                  |S0-2 S1-2 S2-2|
-     *
-     * S1-0 = S[1] at node 0
-     */
+    GeometryType& r_geometry = this->GetGeometry();
+    const double& area       = r_geometry.Area(); // In 3D this is volume
```

Comment:
This can be simplified a little bit (since this function is now changed anyway)
```suggestion
    const auto area       = this->GetGeometry().Area(); // In 3D this is volume
```
Also to me the comment isn't that clear (would Area automatically return volume for a 3D element?)